File: Frontend/angular-app/src/app/core/abstraction/auditable-entity.ts
import {EntityStates} from "../enums/entity-states";

export abstract class AuditableEntity {

  public id:number = 0;
  public ownerRoleId:number = 0;
  public createdById:number = 0;
  public createdAt:Date = new Date(0);
  public modifiedById:number = 0;
  public modifiedAt:Date = new Date(0);
  public isDeleted:boolean = false;
  public entityState:number = EntityStates.Unchanged;

}

File: Backend/Microservices/Admin/Eefa.Admin.Domain/Abstraction/AuditableEntity.cs
using System;
using System.ComponentModel.DataAnnotations;

public abstract class AuditableEntity
{
    [Key]
    public int Id { get; set; }
    public DateTime CreatedAt { get; set; }
    public int? ModifiedById { get; set; }
    public DateTime ModifiedAt { get; set; }
    public bool IsDeleted { get; set; }
    public int OwnerRoleId { get; set; }
    public int CreatedById { get; set; }
}

